### PR TITLE
feat: add strategy all to StrategyCard

### PR DESCRIPTION
**Pull request by mathiusj**
### What problem is this PR solving?

closes https://github.com/Shopify/discount-app-components/issues/179

### Reviewers’ hat-rack :tophat:

- `yarn run storybook`

Before 
image 

After for Product: 
image

After for Shipping:

image

After for Order:

image

### Before you deploy

> **Warning**
> With every PR, you **MUST** think through each of the items in the following checklist and check the appropriate ones. This step cannot be overlooked by the **PR author** or its **reviewers**. Please remove this warning when you're done.

- [x] This PR is safe to rollback.
- [x] I tophatted this change on Storybook.

### CODE REVIEW

**Review by samihibrahim — COMMENTED** [review]
would this change the card for all discount types to include `ALL`? Because it is only supported for Product discounts - both order and shipping shouldn't have `ALL` as an option.

**Comment by mathiusj** [comment]
> would this change the card for all discount types to include `ALL`? Because it is only supported for Product discounts - both order and shipping shouldn't have `ALL` as an option.

hmm good point 🤔 I will look into how to best handle this!

**Comment by mathiusj** [comment]
@aeperea @samihibrahim updated UI to conditionally render specific strategies based on discount class:

Order = First, Maximum
Product = First, Maximum, All
Shipping = All

File: src/components/DiscountApplicationStrategyCard/DiscountApplicationStrategyCard.tsx [hunk 1]
```diff
@@ -17,12 +19,43 @@ const I18N_SCOPE = {
...
 
   const handleChange = (strategies: DiscountApplicationStrategy[]) =>
     strategy.onChange(strategies[0]);
 
+  const isProduct = discountClass === DiscountClass.Product;
+  const isOrder = discountClass === DiscountClass.Order;
+  const isShipping = discountClass === DiscountClass.Shipping;
+
+  const choices = [
+    ...(isProduct || isOrder
```
Comment (samihibrahim) [inline src/components/DiscountApplicationStrategyCard/DiscountApplicationStrategyCard.tsx:34]: ❤️

**Review by aeperea — APPROVED** [review]
LGTM! 🎉

**Pull request merged**